lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
+            self.stop(interaction=itx)
+
+        await msg.edit(content="Set the correct permissions for the verification channel!", view=None)
+        self.stop(interaction=itx)
+
+
+class GatekeeperSetup(UpdateableMenu, ExitableMenu):
+    ctx: AutoModContext
+    record: dict[str, Any]
+
+    def __init__(self, gatekeeper: Optional[GateKeeperConfig] = None, *, context: AutoModContext):
+        super().__init__(context=context,
+                         delete_message_after=True,
+                         timeout=180)
+
+        self.gatekeeper: Optional[GateKeeperConfig] = gatekeeper
+
+    async def format_initial_message(self, ctx: GuildContext):
+        query = "SELECT * FROM guild_gatekeeper_config WHERE guild_id=$1;"
+        record = await ctx.bot.pool.fetchrow(query, ctx.guild.id)
+        self.record = record
+
+        setup_buttons = (self.set_gatekeeper_role, self.set_gatekeeper_channel)
+
+        if record and record['active']:
+            text = "Lightning Gatekeeper is currently active and will gatekeep every new member that joins!\n"\
+                   f"**Verification Role**: {self.gatekeeper.role.mention}\n"\
+                   f"**Verification Channel**: {self.gatekeeper.verification_channel.mention}"
+            for button in setup_buttons:
+                button.disabled = True
+            self.set_switch_labels(True)
+            return text
+
+        self.send_verification_message.disabled = True
+
+        if record is None:
+            text = "Lightning Gatekeeper is not fully set up!"
+            self.set_gatekeeper_role.disabled = False
+            self.set_gatekeeper_channel.disabled = True
+            self.disable_gatekeeper.disabled = True
+            return text
+        elif record['role_id'] is None:
+            text = "Lightning Gatekeeper is not fully set up!"
+            self.disable_gatekeeper.disabled = True
+            self.set_gatekeeper_channel.disabled = True
+            self.set_switch_labels(False)
+            return text
+        elif record['verification_channel_id'] is None:
+            text = "Lightning Gatekeeper is not fully set up!"
+            self.set_gatekeeper_role.disabled = True
+            self.disable_gatekeeper.disabled = True
+            self.set_gatekeeper_channel.disabled = False
+            self.set_switch_labels(False)
+            return text
+        elif record['active'] is False:
+            text = "Lightning Gatekeeper is currently disabled!\n"\
+                   "*Click the Enable button to enable the gatekeeper for everyone*"
+            self.disable_gatekeeper.disabled = False
+            for button in setup_buttons:
+                button.disabled = False
+            self.set_switch_labels(False)
+
+        if record['verification_channel_id'] and record['role_id']:
+            self.disable_gatekeeper.disabled = False
+            self.send_verification_message.disabled = False
+
+        return text
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_clarification):** Consider handling the case where `record` is not found more explicitly.

When `record` is `None`, it indicates that the gatekeeper configuration is not set up. It might be helpful to guide the user on how to set it up or provide a direct action to initiate the setup.

```suggestion
if record is None:
    text = "Lightning Gatekeeper is not fully set up! Use `/setup_gatekeeper` to start the setup process."
    self.set_gatekeeper_role.disabled = False
    self.set_gatekeeper_channel.disabled = True
    self.disable_gatekeeper.disabled = True
    return text
```